lib/ASN1/Universal/Integer.php

```diff
@@ -42,15 +42,20 @@ public function getContent()
         return $this->value;
     }
 
+    public function rightShift($number, $positions)
+    {
+        // Shift 1 right = div / 2
+        return gmp_strval(gmp_div($number, gmp_pow(2, (int)$positions)));
+    }
+
     protected function calculateContentLength()
     {
         $nrOfOctets = 1; // we need at least one octet
-        $tmpValue = abs($this->value);
-        while ($tmpValue > 127) {
-            $tmpValue = $tmpValue >> 8;
+        $tmpValue = gmp_abs(gmp_init($this->value));
```

Comment:
According to [the documentation][1] you don't need the `gmp_init` here.

> Note:
It is not necessary to call this function if you want to use integer or string in place of GMP number in GMP functions, like gmp_add(). Function arguments are automatically converted to GMP numbers, if such conversion is possible and needed, using the same rules as gmp_init().

[1]: http://php.net/manual/en/function.gmp-init.php